### PR TITLE
Build Krew archive and manifest

### LINKED ISSUE #1030: HNC: Use Krew (attempt two)
/assign @adrianludwin

### PR DESCRIPTION
**Pull request by adrianludwin**
Tested: `make krew-install` and verified that the plugin worked
correctly and the documentation link was correct. Built a test release
to my personal repo and verified that I could download the manifest and
install from it correctly.

Fixes #1030

### CODE REVIEW

**Comment by adrianludwin** [comment]
NB all artifacts from my test build are at https://github.com/adrianludwin/multi-tenancy/releases/tag/hnc-v0.6.0-test2

File: incubator/hnc/Makefile [hunk 1]
```diff
@@ -352,3 +332,26 @@ endif
...
 	$(error The image ${HNC_RELEASE_IMG} already exists. Force and overwrite this image by using HNC_FORCE_RELEASE=true)
 endif
+
+# Generate the Krew manifest and put it in manifests/. Note that 'manifests' must exist because
+# krew-build calls krew-tar calls build calls manifests.
+krew-build: krew-tar
+	cp hack/krew-kubectl-hns.yaml manifests/krew-kubectl-hns.yaml
+	sed -i 's/HNC_KREW_TAR_SHA256/${HNC_KREW_TAR_SHA256}/' manifests/krew-kubectl-hns.yaml
+	sed -i 's/HNC_IMG_TAG/${HNC_IMG_TAG}/' manifests/krew-kubectl-hns.yaml
+	sed -i 's/HNC_RELEASE_REPO_OWNER/${HNC_RELEASE_REPO_OWNER}/' manifests/krew-kubectl-hns.yaml
```
Comment (yiqigao217) [inline incubator/hnc/Makefile:342]: Question: what does `'s/HNC_KREW_TAR_SHA256/${HNC_KREW_TAR_SHA256}/'` mean specificially?

File: incubator/hnc/Makefile [hunk 2]
```diff
@@ -104,29 +107,6 @@ kubectl: build
...
-	tar -zcvf bin/kubectl-hns.tar.gz bin/kubectl
-
-# Install kubectl plugin locally using krew.
-krew-install: krew-build
-	kubectl krew install --manifest=manifests/krew-hierarchical-namespaces.yaml --archive=bin/kubectl-hns.tar.gz
-
-# Uninstall kubectl plugin locally using krew.
-krew-uninstall:
-	-kubectl krew uninstall hierarchical-namespaces
-
```
Comment (yiqigao217) [inline incubator/hnc/Makefile:129]: What's the difference between the old and the new?

File: incubator/hnc/hack/krew-kubectl-hns.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: hns
+spec:
+  shortDescription: Manipulate hierarchical namespaces provided by HNC.
```
Comment (ahmetb) [inline incubator/hnc/hack/krew-kubectl-hns.yaml:6]: something as short as:

> Manage hierarchical namespaces

can be better.

**Review by ahmetb — COMMENTED** [review]
Looks good to me.

File: incubator/hnc/Makefile (hunk 1 above)
Comment (adrianludwin) [reply incubator/hnc/Makefile:342]: It means "replace the string `HNC_KREW_TAR_SHA256` with the contents of the variable that happens to have the same name." `sed -i` does an in-place search-and-replace using regexes.

File: incubator/hnc/Makefile (hunk 2 above)
Comment (adrianludwin) [reply incubator/hnc/Makefile:129]: `krew-tar` now includes the license file, while `krew-build` replaces more parts of the template file.

File: incubator/hnc/hack/krew-kubectl-hns.yaml (hunk 3 above)
Comment (adrianludwin) [reply incubator/hnc/hack/krew-kubectl-hns.yaml:6]: sg - I want to keep the term "HNC" in there somewhere but I'll shorten the rest.

**Comment by rjbez17** [comment]
lgtm after the nits.

File: incubator/hnc/hack/krew-kubectl-hns.yaml (hunk 3 above)
Comment (adrianludwin) [reply incubator/hnc/hack/krew-kubectl-hns.yaml]: Shortened to "Manage hierarchical namespaces (part of HNC)"

**Comment by adrianludwin** [comment]
> lgtm after the nits.

Nits should now be addressed :)

**Pull request merged**